sr_module.c

```diff
@@ -199,28 +199,48 @@ int register_module(const struct module_exports* e, char* path, void* handle)
 
 static inline int version_control(const struct module_exports* exp, char *path)
 {
-	if ( !exp->version ) {
+	const char *hint = "(try `make clean all modules' and reinstall everything)";
+	if ( !exp->ver_info.version ) {
 		LM_CRIT("BUG - version not defined in module <%s>\n", path );
 		return 0;
 	}
-	if ( !exp->compile_flags ) {
+	if ( !exp->ver_info.compile_flags ) {
 		LM_CRIT("BUG - compile flags not defined in module <%s>\n", path );
 		return 0;
 	}
+	if ( !exp->ver_info.scm.type ) {
+		LM_CRIT("BUG - SCM type not defined in module <%s> %s\n", path, hint );
+		return 0;
+	}
+	if ( !exp->ver_info.scm.rev ) {
+		LM_CRIT("BUG - SCM revision not defined in module <%s> %s\n", path, hint );
+		return 0;
+	}
 
-	if (strcmp(OPENSIPS_FULL_VERSION, exp->version)==0){
-		if (strcmp(OPENSIPS_COMPILE_FLAGS, exp->compile_flags)==0)
-			return 1;
-		else {
-			LM_ERR("module compile flags mismatch for %s "
-				" \ncore: %s \nmodule: %s\n",
-				exp->name, OPENSIPS_COMPILE_FLAGS, exp->compile_flags);
-			return 0;
-		}
+	if (strcmp(OPENSIPS_FULL_VERSION, exp->ver_info.version)!=0) {
+		LM_ERR("module version mismatch for %s; core: %s; module: %s\n",
```

Comment:
All these logs about mismatch should be CRIT (as it prevents the startup), rather then the regular ERR... imho